pancancer_evaluation/utilities/analysis_utilities.py

```diff
@@ -0,0 +1,79 @@
+import os
+import sys
+
+import numpy as np
+import pandas as pd
+# TODO should this be a paired t-test?
+from scipy.stats import ttest_ind
+
+def load_prediction_results(results_dir, train_set_descriptor):
+    """Load results of mutation prediction experiments.
+
+    Arguments
+    ---------
+    results_dir (str): directory to look in for results, subdirectories should
+                       be experiments for individual genes
+
+    Returns
+    -------
+    results_df (pd.DataFrame): results of classification experiments
+    """
+    results_df = pd.DataFrame()
+    for gene_name in os.listdir(results_dir):
+        gene_dir = os.path.join(results_dir, gene_name)
+        if not os.path.isdir(gene_dir): continue
+        for results_file in os.listdir(gene_dir):
+            if 'classify' not in results_file: continue
+            full_results_file = os.path.join(gene_dir, results_file)
+            gene_results_df = pd.read_csv(full_results_file, sep='\t')
+            gene_results_df['train_set'] = train_set_descriptor
+            gene_results_df['identifier'] = (
+                gene_results_df['gene'] + '_' +
+                gene_results_df['holdout_cancer_type']
+            )
+            results_df = pd.concat((results_df, gene_results_df))
+    return results_df
+
+def compare_results(results_df,
+                    identifier='gene',
+                    metric='auroc',
+                    correction=False,
+                    correction_method='fdr_bh',
+                    correction_alpha=0.05,
+                    verbose=False):
+    """which gene/cancer type combinations beat the negative control baseline?
+
+    TODO better documentation
+    """
+    results = []
+    unique_identifiers = np.unique(results_df[identifier].values)
+    for id_str in unique_identifiers:
+        signal_results = results_df[
+            (results_df[identifier] == id_str) &
+            (results_df.data_type == 'test') &
+            (results_df.signal == 'signal')
+        ][metric].values
+        shuffled_results = results_df[
+            (results_df[identifier] == id_str) &
+            (results_df.data_type == 'test') &
+            (results_df.signal == 'shuffled')
+        ][metric].values
```

Comment:
You can avoid having a hanging `[metric].values` by declaring the conditions ahead of time like in the code below. It's just a style choice, but seeing a right bracket on a newline confused me for a second. (If you decide to use the suggestion, double check it. I always screw up multiple conditions in pandas the first time)

```suggestion
    conditions = ((results_df[identifier] == id_str) &
        (results_df.data_type == 'test') &
        (results_df.signal == 'signal')
        )
    signal_results = results_df[conditions][metric].values
    
    conditions = ((results_df[identifier] == id_str) &
         (results_df.data_type == 'test') &
         (results_df.signal == 'shuffled')
         )
    shuffled_results = results_df[conditions][metric].values
```

Reply:
Good idea! I like your approach better, it's definitely cleaner IMO.